src/appshell/internal/applicationactioncontroller.cpp

```diff
@@ -19,16 +19,43 @@
 #include "applicationactioncontroller.h"
 
 #include <QCoreApplication>
+#include <QMainWindow>
 
 using namespace mu::appshell;
 using namespace mu::actions;
 
 void ApplicationActionController::init()
 {
     dispatcher()->reg(this, "quit", this, &ApplicationActionController::quit);
+
+    dispatcher()->reg(this, "fullscreen", this, &ApplicationActionController::toggleFullScreen);
+}
+
+mu::ValCh<bool> ApplicationActionController::isFullScreen() const
+{
+    ValCh<bool> result;
+    result.ch = m_fullScreenChannel;
+    result.val = mainWindow()->qMainWindow() ? mainWindow()->qMainWindow()->isFullScreen() : false;
```

Comment:
Better to move this code to a separate method because you use it in 2 different places:

bool isFullScreen() {
return mainWindow()->qMainWindow() ? mainWindow()->qMainWindow()->isFullScreen() : false;
}